source/dub/compilers/utils.d

```diff
@@ -266,16 +266,28 @@ private enum probeEndMark = "__dub_probe_end__";
 
 	See_Also: `readPlatformProbe`
 */
-NativePath generatePlatformProbeFile()
+NativePath generatePlatformProbeFile(bool betterC = false)()
 {
 	import dub.internal.vibecompat.core.file;
 	import dub.internal.vibecompat.data.json;
 	import dub.internal.utils;
 	import std.string : format;
+	enum moduleInfo = betterC ? q{
+		module object;
+		alias string = const(char)[];
+		alias size_t = uint;
+		template _d_arrayappendcTXImpl(Tarr : T[], T){
+			ref Tarr _d_arrayappendcTX(return ref scope Tarr px, size_t n) @trusted pure nothrow{return px;}
+		}
+		const(char)[][] _d_arrayappendcTX(return scope ref const(char)[][] px, size_t n){return px;}
+		Tarr _d_arrayappendT(Tarr : T[], T)(return ref scope Tarr x, scope Tarr y){return x;}
+	} : q{
+		module dub_platform_probe;
+	};
```

Comment:
I'm a bit concerned about this part, as it seems to be specific to the current state of BetterC. Future improvements might make this out of date. Am I right that the problem arises because the platform probe has a dependency on the runtime ? Couldn't we achieve something similar with a sufficiently sized static array ?

Reply:
It's the string[] appender running as CTFE that calls the runtime as a stub, if there's any improvements we would be removing some definitions and that's pretty much it. Either way if anything breaks it'll show a warning and fallback to using the runtime (through the bool retry in the catch)

Reply:
The CTFE doesn't support using static storage either, you have to use the GC to create an allocation and then you can reference it in scope

Reply:
I managed to remove the declarations by using only pragma(msg, ) and formatting the string at the end